File: gommo-backend/src/main/java/br/com/gommo/modules/person/performance/mapper/PerformanceReviewMapper.java
package br.com.gommo.modules.person.performance.mapper;
import br.com.gommo.modules.person.performance.dto.*;
import br.com.gommo.modules.person.performance.entity.PerformanceReview;
import org.springframework.stereotype.Component;
@Component public class PerformanceReviewMapper {
    public PerformanceReview toEntity(PerformanceReviewRequestDto dto) {
        return PerformanceReview.builder().collaboratorId(dto.getCollaboratorId())
            .periodStart(dto.getPeriodStart()).periodEnd(dto.getPeriodEnd()).rating(dto.getRating())
            .goalsSummary(dto.getGoalsSummary()).feedback(dto.getFeedback()).reviewerName(dto.getReviewerName()).build();
    }
    public void updateEntity(PerformanceReview entity, PerformanceReviewRequestDto dto) {
        entity.setCollaboratorId(dto.getCollaboratorId());
        entity.setPeriodStart(dto.getPeriodStart());
        entity.setPeriodEnd(dto.getPeriodEnd());
        entity.setRating(dto.getRating());
        entity.setGoalsSummary(dto.getGoalsSummary());
        entity.setFeedback(dto.getFeedback());
        entity.setReviewerName(dto.getReviewerName());
    }
    public PerformanceReviewResponseDto toResponse(PerformanceReview entity) {
        return PerformanceReviewResponseDto.builder().id(entity.getId()).code(entity.getCode()).status(entity.getStatus())
            .collaboratorId(entity.getCollaboratorId()).periodStart(entity.getPeriodStart())
            .periodEnd(entity.getPeriodEnd()).rating(entity.getRating()).goalsSummary(entity.getGoalsSummary())
            .feedback(entity.getFeedback()).reviewerName(entity.getReviewerName())
            .createdAt(entity.getCreatedAt()).updatedAt(entity.getUpdatedAt()).build();
    }
}

File: gommo-frontend/src/modules/person/performance/lib/performance-review.mapper.ts
import type {
    PerformanceReview,
    PerformanceReviewCreateDto,
} from "@/modules/person/performance/dto/performance-review.dto";

export function performanceReviewToFormDto(entity: PerformanceReview): PerformanceReviewCreateDto {
    return {
        collaboratorId: entity.collaboratorId ?? "",
        periodStart: entity.periodStart?.slice(0, 10) ?? "",
        periodEnd: entity.periodEnd?.slice(0, 10) ?? "",
        rating: entity.rating,
        goalsSummary: entity.goalsSummary ?? "",
        feedback: entity.feedback ?? "",
        reviewerName: entity.reviewerName ?? "",
    };
}

export const emptyPerformanceReviewForm = (): PerformanceReviewCreateDto => ({
    collaboratorId: "",
    periodStart: "",
    periodEnd: "",
    goalsSummary: "",
    feedback: "",
    reviewerName: "",
});
